colcon_core/task/python/build.py

```diff
@@ -105,6 +105,11 @@ async def build(self, *, additional_hooks=None):  # noqa: D102
             self._append_install_layout(args, cmd)
             completed = await run(
                 self.context, cmd, cwd=args.build_base, env=env)
+
+            # Remove symlinks that were only needed during build time
```

Comment:
Should this happen in a `finally` block instead to ensure it is done even if the in-between code raises an exception?

Reply:
Fixed, but accidentally squashed fix during a rebase. Latest state does it in a finally block.